library/psa_crypto.c

```diff
@@ -2399,6 +2436,7 @@ psa_status_t psa_mac_sign_finish( psa_mac_operation_t *operation,
         memset( &mac[operation->mac_size], '!',
                 mac_size - operation->mac_size );
 
+cleanup:
```

Comment:
This will change behaviour for the error case (it will fill the buffer, instead of leaving it unchanged as is currently done), although this is within spec & isn't API-breaking (so could go into 2.x). I'll go ahead with this for now, but see what @gilles-peskine-arm thinks.

Reply:
Good point Ronald, setting the buffer to something that cannot be a valid MAC should preferably be done on all error paths.